strictdoc/core/traceability_index.py

```diff
@@ -537,6 +541,61 @@ def parent_cycle_traverse_(node_id: str) -> List[Any]:
                 datetime.datetime.today(),
             )
 
+    def update_node_mid(self, node: SDocNode, new_mid: str) -> None:
+        """
+        Update a node’s MID identifier and update all parent and child node
+        relations to reference this new MID.
+
+        The graph database contains relations between nodes. Since these
+        relations remain unchanged, the job of this function is only to update
+        each node's original .relations, so that they point to the new MID.
+        """
+
+        old_mid = node.reserved_mid
+
+        #
+        # Update all child nodes.
+        #
+        child_nodes: OrderedSet[SDocNode] = self.graph_database.get_link_values(
+            link_type=GraphLinkType.NODE_TO_CHILD_NODES,
+            lhs_node=node,
+            edge=ALL_EDGES,
+        )
+        for child_node_ in child_nodes:
+            for relation_ in copy(child_node_.relations):
+                if (
+                    isinstance(relation_, ParentReqReference)
+                    and relation_.ref_uid == old_mid
```

Comment:
Can we (hypothetically, very unlikely) run into a case where
- doc1 has `RELATION_FIELD: MID` and a requirement with MID "123"
- doc2 has `RELATION_FIELD: UID` and a requirement with UID "123"
- some `relation_.ref_uid` happens to mean UID "123", not MID "123?

Could we distinguish if a reference means MID "123" instead of UID "123"? If it's easy, maybe add the check here.

Reply:
I have added the `config.relation_field != "MID":` check. I think it covers the case you described.